packages/expo-file-system/src/FileSystem.ts

```diff
@@ -271,49 +399,29 @@ export class DownloadResumable {
     if (!ExponentFileSystem.downloadResumableStartAsync) {
       throw new UnavailabilityError('expo-file-system', 'downloadResumableStartAsync');
     }
-    this._addSubscription();
+
+    if (this.checkIfTaskWasCanceled()) {
+      return undefined;
```

Comment:
That's the common way to return `undefined` 😉 
```suggestion
      return;
```